pkg/nodeagent/apis/config/types.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeAgentConfiguration defines the configuration for the gardener-node-agent.
+type NodeAgentConfiguration struct {
+	metav1.TypeMeta
+
+	// APIServer contains the connection configuration for the gardener-node-agent to
+	// access the shoot api server.
+	APIServer APIServer
+
+	// SecretName defines the name of the secret in the shoot cluster control plane, which contains
+	// the OSC for the gardener-node-agent.
+	OSCSecretName string
```

Comment:
Let's not use "OSC" abbreviation in the API. 
```suggestion
	OperatingSystemConfigSecretName string
```

same in versioned file